network-api/networkapi/wagtailpages/tests/buyersguide/test_article_page.py

```diff
@@ -206,3 +210,112 @@ def test_secondary_related_articles_no_related_articles(self):
         result = article_page.get_secondary_related_articles()
 
         self.assertListEqual(result, [])
+
+    @staticmethod
+    def generate_form_data(article_page, data: dict) -> dict:
+        """
+        Generate valid form data for the BuyersGuide article page form.
+
+        We add the extra fields below to the dict as all these fields are set as
+        required, and the form will return a validation error against them if left out.
+
+        Since we have the "body" field set as required, we need to submit at least
+        one StreamField in the form, or else the form will return a validation error.
+        """
+        return nested_form_data(
+            {
+                **data,
+                "title": article_page.title,
+                "slug": article_page.slug,
+                "body": streamfield([("paragraph", rich_text("Hello world!"))]),
+            }
+        )
+
+    def test_article_page_requires_search_image(self):
+        """
+        Test that the buyersguide article page form will
+        return a validation error if no "search_image" is set.
+        """
+        article_page = buyersguide_factories.BuyersGuideArticlePageFactory(parent=self.content_index)
+
+        test_form = self.article_page_form(
+            instance=article_page,
```

Comment:
Can you add some context why we need an existing `instance`? The form should be working in a create scenario too.

Reply:
Hi Tibor, no reason in particular. I just added this to try and test the event of "someone is editing an existing articlepage". However, after giving it some more thought, I believe that a create scenario works better, and also makes the code a little less confusing, so I have decided to remove the instance!